pilot/pkg/model/push_context.go

```diff
@@ -280,6 +280,9 @@ type PushRequest struct {
 	// The kind of resources are defined in pkg/config/schemas.
 	ConfigsUpdated map[ConfigKey]struct{}
 
+	// UpdateTypes tracks the action (add, update, delete) that was taken on a configuration.
+	UpdateTypes map[ConfigKey]Event
```

Comment:
I think we don't need this. For Service, if the Service doesn't exist (ie SidecarScope.Service returns `nil`), then we know the cluster is not needed, and we can remove it.

For other resources, like DR, whether its added/removed/updated doesn't matter at all. We just regenerate the clusters and send them out.

Reply:
I see -- I will make it so `service=nil` means it was deleted